# HG changeset patch
# User Rik <rik@octave.org>
# Date 1475281055 25200
#      Fri Sep 30 17:17:35 2016 -0700
# Branch stable
# Node ID 6b2a1b2c1576bfdbedc34bb70db03ca91844bbaf
# Parent  d5e775944809586d4753eb7139bdd4bad572adac
__gnuplot_draw_axes__.m: Handle case where clim range is 0.

* __gnuplot_draw_axes__.m (mapcdata): If clim range (clim(2) - clim(1)) is
zero, then all cdata takes on the maximum possible value.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -2850,28 +2850,39 @@ function maybe_do_x2tick_mirror (plot_st
                           axis_obj.tickdir);
   endif
 
 endfunction
 
 function retval = mapcdata (cdata, mode, clim, cmap_sz)
   if (ndims (cdata) == 3)
     ## True Color, clamp data to 8-bit
+    clim = double (clim);
     cdata = double (cdata);
-    cdata = 255 * (cdata - clim(1)) / (clim(2)-clim(1));
-    cdata(cdata < 0) = 0;  cdata(cdata > 255) = 255;
+    clim_rng = clim(2) - clim(1);
+    if (clim_rng != 0)
+      cdata = 255 * (cdata - clim(1)) / clim_rng;
+      cdata(cdata < 0) = 0;  cdata(cdata > 255) = 255;
+    else
+      cdata(:) = 255;
+    endif
     ## Scale using inverse of gnuplot's cbrange mapping
     retval = 1 + cdata * (cmap_sz-1)/255;
   else
     if (islogical (cdata))
       cdata += 1;
     elseif (strcmp (mode, "scaled"))
+      clim = double (clim);
       cdata = double (cdata);
-      clim = double (clim);
-      cdata = 1 + fix (cmap_sz * (cdata - clim(1)) / (clim(2) - clim(1)));
+      clim_rng = clim(2) - clim(1);
+      if (clim_rng != 0)
+        cdata = 1 + fix (cmap_sz * (cdata - clim(1)) / clim_rng);
+      else
+        cdata(:) = cmap_sz;
+      endif
     else
       if (isinteger (cdata))
         cdata += 1;
       else
         cdata = fix (cdata);
       endif
     endif
     retval = max (1, min (cdata, cmap_sz));
